components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/FileBlobStore.java

```diff
@@ -307,6 +307,11 @@ public void init(final BlobStoreConfiguration configuration) throws IOException
     this.metadataStore = MapdbBlobMetadataStore.create(metadataFile);
   }
 
+  @Override
+  public AutoClosableIterable<BlobId> iterator() {
+    return metadataStore.findWithState(BlobState.ALIVE);
```

Comment:
why only alive?  and/or why not expose the state to iterate?

asking because one use of this is for admin/debug so it may be useful to have the ability to see all the blobs, not just those that are alive.